core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -1786,4 +1786,32 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite with PrivateMethodT
       ShuffleBlockId(0, 5, 2), ShuffleBlockId(0, 6, 2)))
   }
 
+  test("SPARK-38987: failure to fetch corrupted shuffle block chunk should " +
```

Comment:
Nit: modify this name so it's clear that when corruption goes undetected then it should throw fetch failure. Otherwise, it is confusing because one test says we should fallback on corruption and the other says it should throw fetch failure